### PR TITLE
WD-12223 Refreshed /data/opensearch

### PR DESCRIPTION
**Pull request by immortalcodes**
## Done

- Refreshed /data/opensearch
- This is part of many pages that needs to be created under /data/opensearch , hence some of navigation links might return 404.
## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: http://0.0.0.0:8002/
- Run through the following [QA steps](https://discourse.canonical.com/t/qa-steps/152)
- [Demo](https://canonical-com-1308.demos.haus/data/opensearch)
- [Copydoc](https://docs.google.com/document/d/1nVdoKoPrXw8lsvHEtMichlPNfxYU9Uj1zmKowcUXQ_w/edit)
- [Figma file](https://www.figma.com/design/uJUwbUAGDog4KCMivrMpnl/24.10-Data-Fabric?node-id=10-10158&t=hD5i7YosKZ9Ru1dD-0)

## Issue / Card

Fixes # [WD-12223](https://warthogs.atlassian.net/browse/WD-12223)

[WD-12223]: https://warthogs.atlassian.net/browse/WD-12223?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: templates/data/opensearch.html [hunk 1]
```diff
@@ -1,345 +1,500 @@
...
+            </p>
+            <hr class="p-rule is-muted" />
+            <div class="p-section--shallow  u-no-margin--bottom">
+              <ul class="p-list--divided">
+                <li class="p-list__item is-ticked">BExpert troubleshooting on hand</li>
+                <li class="p-list__item is-ticked">Security and bug fixes</li>
+                <li class="p-list__item is-ticked">Full stack coverage</li>
+              </ul>
+              <br />
+              <br />
```
Comment (mtruj013) [inline templates/data/opensearch.html:365]:
```suggestion
```
Is there a better solution we can use here? [br elements](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/br) should be used to add breaks in text, not to manipulate spacing. Could you have a look at the vanilla docs?

File: templates/data/opensearch.html [hunk 2]
```diff
@@ -1,345 +1,500 @@
...
+        </div>
+      </div>
+    </section>
+
+    <hr class="is-fixed-width" />
+
+    <section class="p-strip is-deep">
+      <div class="u-fixed-width">
+        <p class="p-heading--2">
+          <a href="https://ubuntu.com/data/docs/opensearch/iaas">Updates from the product manager&nbsp;&rsaquo;</a>
```
Comment (mtruj013) [inline templates/data/opensearch.html:443]:
```suggestion
          <a href="/blog/author/mtabirao">Hear from the product manager&nbsp;&rsaquo;</a>
```

File: templates/data/opensearch.html [hunk 3]
```diff
@@ -1,345 +1,500 @@
...
-</section>
-
-<section class="p-section">
-  <div class="row">
-    <hr class="p-rule" />
-    <div class="col-6">
-      <p>OpenSearch is a registered trademark of Amazon Web Services. Other trademarks are property of their respective owners. Charmed OpenSearch is not sponsored, endorsed, or affiliated with Amazon Web Services.</p>
-    </div>
+    </section>
   </div>
```
Comment (mtruj013) [inline templates/data/opensearch.html:491]:
```suggestion
```

**Review by mtruj013 — CHANGES_REQUESTED** [review]
Thanks @immortalcodes , some comments in the code. I also left some comments on the figma about the muted headings if you could have a look and I believe we're also missing updated assets as the ones in the demo don't match the figma: 
image
vs:
Screenshot from 2024-07-22 13-14-40

**Comment by immortalcodes** [comment]
> Thanks @immortalcodes , some comments in the code. I also left some comments on the figma about the muted headings if you could have a look and I believe we're also missing updated assets as the ones in the demo don't match the figma: image vs: Screenshot from 2024-07-22 13-14-40



> Thanks @immortalcodes , some comments in the code. I also left some comments on the figma about the muted headings if you could have a look and I believe we're also missing updated assets as the ones in the demo don't match the figma: image vs: Screenshot from 2024-07-22 13-14-40

The assets are the ones that are provided in the copydoc so I followed them, shall I give preference to figma assets?
There is comment about same by Sophie as well in the figma.

File: templates/data/opensearch.html [hunk 4]
```diff
@@ -1,345 +1,500 @@
...
-							loading="lazy",
-							attrs={"class": "p-logo-section__logo"}
-							) | safe
-						}}
+      </div>
+      <div class="row--50-50">
+      <p class="col">
+        When you use Canonical’s OpenSearch, you get the benefit of running on top of an operating system image that’s optimised for every cloud. Run your OpenSearch on Azure, GCP, AWS, on Kubernetes or your private data centre, with OpenStack, Microcloud and VMWare.
+      </p>
+      </div>
```
[the remark this answers is not in the document]
Comment (immortalcodes) [reply templates/data/opensearch.html:157]: Nope, it wasn't by design. I know it's not the best layout for the text and I didn't know what to do better

File: templates/data/opensearch.html (hunk 1 above)
Comment (mtruj013) [reply templates/data/opensearch.html]: Adding a `p-section` on `p-card--overlay` div seems to work nicely:
image

File: templates/data/opensearch/index.html [hunk 5]
```diff
@@ -0,0 +1,530 @@
...
+          <div class="p-section--shallow">
+            <h1>OpenSearch for the enterprise</h1>
+            <p>
+              Secure and automate the deployment, maintenance and upgrades of your search and analytics suite across private and public clouds.
+            </p>
+          </div>
+          <hr />
+          <a class="p-button--positive"
+             href="/contact-us"
+             aria-controls="data-spark-modal">Contact us</a>
```
Comment (mtruj013) [inline templates/data/opensearch/index.html:47]: These CTA's are a bit vague so we can add this to improve accessibility for people using screen readers 
```suggestion
             aria-controls="data-spark-modal">Contact us <span class="u-off-screen">about OpenSearch for the enterprise</span></a>
```

File: templates/data/opensearch/index.html [hunk 6]
```diff
@@ -0,0 +1,530 @@
...
+          <a class="p-button"
+             href="https://assets.ubuntu.com/v1/6dc3440a-Charmed%20OpenSearch%20-%20Datasheet%20-%20amended%2004%2006%202024.pdf">Datasheet</a>
```
Comment (mtruj013) [inline templates/data/opensearch/index.html:49]:
```suggestion
             href="https://assets.ubuntu.com/v1/6dc3440a-Charmed%20OpenSearch%20-%20Datasheet%20-%20amended%2004%2006%202024.pdf"><span class="u-off-screen">Download the Charmed OpenSearch</span> Datasheet</a>
```

File: templates/data/opensearch/index.html [hunk 7]
```diff
@@ -0,0 +1,530 @@
...
+        </div>
+      </div>
+    </div>
+  </section>
+
+  <section class="p-section">
+    <hr class="p-rule is-fixed-width" />
+    <div class="row">
+      <div class="col-3 col-medium-2">
+        <h2 class="p-text--small-caps highlight">Why choose Canonical's enterprise-grade OpenSearch?</h2>
```
Comment (mtruj013) [inline templates/data/opensearch/index.html:59]: No need for this one
```suggestion
        <h2 class="p-text--small-caps">Why choose Canonical's enterprise-grade OpenSearch?</h2>
```

File: templates/data/opensearch/index.html [hunk 8]
```diff
@@ -0,0 +1,530 @@
...
+      </div>
+      <div class="col-9 col-medium-4">
+        <ul class="p-list--divided">
+          <li class="p-list__item">
+            <h2>10 years security maintenance and support</h2>
```
Comment (mtruj013) [inline templates/data/opensearch/index.html:64]: These should just be styled as headings, they aren't headings themselves
```suggestion
            <p class="p-heading--2">10 years security maintenance and support</p>
```

File: templates/data/opensearch/index.html [hunk 9]
```diff
@@ -0,0 +1,530 @@
...
+             aria-controls="data-spark-modal">Contact us</a>
+          <a class="p-button"
+             href="https://assets.ubuntu.com/v1/6dc3440a-Charmed%20OpenSearch%20-%20Datasheet%20-%20amended%2004%2006%202024.pdf">Datasheet</a>
+        </div>
+      </div>
+    </div>
+  </section>
+
+  <section class="p-section">
+    <hr class="p-rule is-fixed-width" />
```
Comment (mtruj013) [inline templates/data/opensearch/index.html:56]: Sorry forgot to mention, but `p-rule` is being deprecated, no need to add it moving forward. If you move the hr one line below so it's in the row you can also remove the `is-fixed-width` class

File: templates/data/opensearch/index.html [hunk 10]
```diff
@@ -0,0 +1,530 @@
...
+      <hr class="p-rule" />
+      <div class="col">
+        <h2 class="p-text--small-caps">Run OpenSearch on any public, private or hybrid cloud</h2>
+      </div>
+      <div class="col">
+        <p>
+          When you use Canonical’s OpenSearch, you get the benefit of running on top of an operating system image that’s optimised for every cloud. Run your OpenSearch on Azure, GCP, AWS, on Kubernetes or your private data centre, with OpenStack, Microcloud and VMWare.
+        </p>
+      </div>
+      <div class="p-logo-section u-hide--small">
```
Comment (mtruj013) [inline templates/data/opensearch/index.html:91]: Is hiding the logos on small screens intentional? I don't believe we do this for any of the other pages

File: templates/data/opensearch/index.html [hunk 11]
```diff
@@ -0,0 +1,530 @@
...
+            ) | safe
+            }}
+          </div>
+          <div class="p-logo-section__item">
+            {{ image (
+            url="https://assets.ubuntu.com/v1/a89f2843-microcloud.png",
+            alt="Microcloud",
+            width="394",
+            height="312",
+            hi_def=True,
```
Comment (mtruj013) [inline templates/data/opensearch/index.html:159]: Missing attribute
```suggestion
            hi_def=True,
            loading="lazy",
```

File: templates/data/opensearch/index.html [hunk 12]
```diff
@@ -0,0 +1,530 @@
...
+        </div>
+      </div>
+    </div>
+  </section>
+
+  <hr class="p-rule is-fixed-width" />
+
+  <section class="p-section">
+    <div class="row--50-50">
+      <div class="col">
```
Comment (mtruj013) [inline templates/data/opensearch/index.html:173]:
```suggestion
  <section class="p-section">
    <div class="row--50-50">
      <hr />
      <div class="col">
```

File: templates/data/opensearch/index.html [hunk 13]
```diff
@@ -0,0 +1,530 @@
...
+        <div class="p-image-wrapper">
+          <div class="p-image-container u-hide--small">
+            {{ image (
+            url="https://assets.ubuntu.com/v1/348fac82-Image.png",
+            alt="",
+            width="1800",
+            height="1015",
+            hi_def=True,
+            attrs={"class": "p-image-container__image"},
+            loading="auto|lazy"
```
Comment (mtruj013) [inline templates/data/opensearch/index.html:186]:
```suggestion
            loading="lazy"
```

File: templates/data/opensearch/index.html [hunk 14]
```diff
@@ -0,0 +1,530 @@
...
+        <div class="p-image-wrapper u-hide--small">
+          <div class="p-image-container">
+            {{ image (
+            url="https://assets.ubuntu.com/v1/6f9200b9-1.png",
+            alt="",
+            width="1800",
+            height="1015",
+            hi_def=True,
+            attrs={"class": "p-image-container__image"},
+            loading="auto|lazy"
```
Comment (mtruj013) [inline templates/data/opensearch/index.html:228]:
```suggestion
            loading="lazy"
```

File: templates/data/opensearch/index.html [hunk 15]
```diff
@@ -0,0 +1,530 @@
...
+        <div class="p-image-wrapper u-hide--small">
+          <div class="p-image-container">
+            {{ image (
+            url="https://assets.ubuntu.com/v1/3065553d-2.png",
+            alt="",
+            width="1800",
+            height="1015",
+            hi_def=True,
+            attrs={"class": "p-image-container__image"},
+            loading="auto|lazy"
```
Comment (mtruj013) [inline templates/data/opensearch/index.html:260]:
```suggestion
            loading="lazy"
```

File: templates/data/opensearch/index.html [hunk 16]
```diff
@@ -0,0 +1,530 @@
...
+        <div class="p-image-wrapper u-hide--small">
+          <div class="p-image-container">
+            {{ image (
+            url="https://assets.ubuntu.com/v1/e907127d-3.png",
+            alt="",
+            width="1800",
+            height="1014",
+            hi_def=True,
+            attrs={"class": "p-image-container__image"},
+            loading="auto|lazy"
```
Comment (mtruj013) [inline templates/data/opensearch/index.html:296]:
```suggestion
            loading="lazy"
```

File: templates/data/opensearch/index.html [hunk 17]
```diff
@@ -0,0 +1,530 @@
...
+    <hr class="p-rule is-fixed-width" />
+    <div class="row">
+      <div class="col-6 col-medium-3">
+        <h2>Charmed OpenSearch services</h2>
+      </div>
+      <div class="col-6 col-medium-3">
+        <p>
+          Charmed OpenSearch is available self-managed or as a fully managed solution remotely delivered in your on-premise environment or cloud tenancy.
+        </p>
+      </div>
```
Comment (mtruj013) [inline templates/data/opensearch/index.html:346]: Please do the same for the cards, row--50-50. This should fix the spacing as right now the text is too close to the cards
```suggestion
    <div class="row--50-50">
      <div class="col">
        <h2>Charmed OpenSearch services</h2>
      </div>
      <div class="col">
        <p>
          Charmed OpenSearch is available self-managed or as a fully managed solution remotely delivered in your on-premise environment or cloud tenancy.
        </p>
      </div>
```

File: templates/data/opensearch/index.html [hunk 18]
```diff
@@ -0,0 +1,530 @@
...
+        <p>
+          OpenSearch is an open-source search and analytics suite. Developers build solutions for search, data observability, data ingestion and more.
+        </p>
+        <hr class="is-muted" />
+        <p>
+          <a href="/data/opensearch/what-is-opensearch">Learn more about OpenSearch&nbsp;&rsaquo;</a>
+        </p>
+      </div>
+    </div>
+    <div class="row--25-75 is-split-on-medium">
```
Comment (mtruj013) [inline templates/data/opensearch/index.html:419]: This class has been deprecated, it's responsive by default now
```suggestion
    <div class="row--25-75">
```

File: templates/data/opensearch/index.html [hunk 19]
```diff
@@ -0,0 +1,530 @@
...
+          </p>
+        </div>
+        <hr class="p-rule is-muted" />
+        <div class="p-section--shallow">
+          <h3 class="p-heading--5 u-no-margin--bottom">
+            <a href="https://ubuntu.com/engage/vector-databases-generative-ai">Vector databases for generative AI applications</a>
+          </h3>
+          <p>Improving the relevance of your LLM application by leveraging Charmed Opensearch’s vector database</p>
+        </div>
+        <hr class="p-rule" />
```
Comment (mtruj013) [inline templates/data/opensearch/index.html:501]:
```suggestion
        <hr class="is-muted" />
```

**Review by mtruj013 — APPROVED** [review]
Thanks @immortalcodes, some changes needed but good to go once those have been addressed

File: templates/data/opensearch/index.html (hunk 17 above)
Comment (immortalcodes) [reply templates/data/opensearch/index.html:346]: There is a nesting of rows below so this format might create issues and how does it adds more space?

File: templates/data/opensearch/index.html [hunk 20]
```diff
@@ -0,0 +1,528 @@
...
+  is-paper
+{% endblock body_class %}
+
+{% block content %}
+  <section class="p-section--hero">
+    <div class="row">
+      <div class="col-3 col-medium-2">
+        <div class="u-hide--small u-vertically-center u-aspect-ratio--16-9"
+             style="margin-top: -55px">
+          {{ image (
```
Comment (akbarkz) [inline templates/data/opensearch/index.html:25]: please remove space between `image` and the bracket and reformat, and do the same throughout the whole file

File: templates/data/opensearch/index.html [hunk 21]
```diff
@@ -0,0 +1,506 @@
+{% extends 'base_index.html' %}
+
+{% block meta_copydoc %}
+  https://docs.google.com/document/d/1nVdoKoPrXw8lsvHEtMichlPNfxYU9Uj1zmKowcUXQ_w/edit
+{% endblock meta_copydoc %}
+
+{% block title %}An enterprise  solution for OpenSearch from Canonical{% endblock %}
+
+{% block canonical_url %}{{ request.host_url }}data/opensearch{% endblock %}
```
Comment (anthonydillon) [inline templates/data/opensearch/index.html:9]: What's your for? Isn't the canonical URL this page?

File: templates/data/opensearch/index.html [hunk 22]
```diff
@@ -0,0 +1,506 @@
...
+{% endblock body_class %}
+
+{% block content %}
+  <section class="p-section--hero">
+    <div class="row">
+      <div class="col-3 col-medium-2">
+        <div class="u-hide--small u-vertically-center u-aspect-ratio--16-9"
+             style="margin-top: -55px">
+          {{ image(url="https://assets.ubuntu.com/v1/6deb3921-opensearch-logo.png",
+                    alt="OpenSearch",
```
Comment (anthonydillon) [inline templates/data/opensearch/index.html:26]: This is not a descriptive alt content

File: templates/data/opensearch/index.html [hunk 23]
```diff
@@ -0,0 +1,506 @@
...
+      <div class="col">
+        <p>
+          When you use Canonical’s OpenSearch, you get the benefit of running on top of an operating system image that’s optimised for every cloud. Run your OpenSearch on Azure, GCP, AWS, on Kubernetes or your private data centre, with OpenStack, Microcloud and VMWare.
+        </p>
+      </div>
+      <div class="p-logo-section">
+        <div class="p-logo-section__items">
+          <div class="p-logo-section__item">
+            {{ image(url="https://assets.ubuntu.com/v1/36a0cd3b-microsoft-azure-2021.png",
+                        alt="Microsoft Azure",
```
Comment (anthonydillon) [inline templates/data/opensearch/index.html:91]: We should mention this is a logo

File: templates/data/opensearch/index.html [hunk 24]
```diff
@@ -0,0 +1,506 @@
...
+          <ul class="p-list--divided">
+            <li class="p-list__item is-ticked">Automated security and configuration best practices built in</li>
+            <li class="p-list__item is-ticked">Made for physical and virtual machines (VMs)</li>
+            <li class="p-list__item is-ticked">Integrated observability for monitoring, alerting and auditing</li>
+            <li class="p-list__item is-ticked">Multi-cloud support</li>
+            <li class="p-list__item is-ticked">Self-healing</li>
+          </ul>
+          <hr class="is-muted" />
+          <p>
+            <a href="/data/opensearch/managed">Explore Managed OpenSearch Solutions&nbsp;&rsaquo;</a>
```
Comment (anthonydillon) [inline templates/data/opensearch/index.html:187]: This link 404s

File: templates/data/opensearch/index.html [hunk 25]
```diff
@@ -0,0 +1,506 @@
...
+                        loading="lazy") | safe
+            }}
+          </div>
+        </div>
+        <p>
+          We provide 10 years of support and security patching for your OpenSearch artefacts in the format of your choice through our Ubuntu Pro subscription.
+        </p>
+        <hr class="is-muted" />
+        <p>
+          <a class="p-button--positive"  href="/data/opensearch#get-in-touch">Contact Us</a>
```
Comment (anthonydillon) [inline templates/data/opensearch/index.html:218]: Please use the provoke-model class with the href going to a real contract us page

File: templates/data/opensearch/index.html [hunk 26]
```diff
@@ -0,0 +1,506 @@
...
+                        loading="lazy") | safe
+            }}
+          </div>
+        </div>
+        <p>
+          Get comprehensive, 24/7 or weekday support for Charmed OpenSearch, covering the entire solution for up to 10 years.
+        </p>
+        <hr class="is-muted" />
+        <p>
+          <a href="/data/opensearch/support">Explore Opensearch Support&nbsp;&rsaquo;</a>
```
Comment (anthonydillon) [inline templates/data/opensearch/index.html:248]: 404

File: templates/data/opensearch/index.html [hunk 27]
```diff
@@ -0,0 +1,506 @@
...
+        <div class="p-section--shallow">
+          <h3 class="p-heading--5 u-no-margin--bottom">
+            <a href="https://ubuntu.com/engage/vector-databases-generative-ai">Vector databases for generative AI applications</a>
+          </h3>
+          <p>Improving the relevance of your LLM application by leveraging Charmed Opensearch’s vector database</p>
+        </div>
+        <hr class="is-muted" />
+        <div class="p-section--shallow">
+          <h3 class="p-heading--5 u-no-margin--bottom">
+            <a href="/data/opensearch#get-in-touch">Discuss your needs with our team</a>
```
Comment (anthonydillon) [inline templates/data/opensearch/index.html:480]: Same here, can we use the class to provoke the model and have this like to a static contract us please

**Review by anthonydillon — CHANGES_REQUESTED** [review]
A few small suggests in the code

File: templates/data/opensearch/index.html (hunk 24 above)
Comment (immortalcodes) [reply templates/data/opensearch/index.html:187]: This page is still in development hence this branch merges to feature and not to main

File: templates/data/opensearch/index.html (hunk 26 above)
Comment (immortalcodes) [reply templates/data/opensearch/index.html:248]: This page is still in development hence this branch merges to feature and not to main

File: templates/data/opensearch/index.html (hunk 21 above)
Comment (immortalcodes) [reply templates/data/opensearch/index.html:9]: Sorry I think I don't fully understand.

Comment (akbarkz) [reply templates/data/opensearch/index.html:9]: I think the question is why do you need this block? isn't it already clear what the URL is?

Comment (immortalcodes) [reply templates/data/opensearch/index.html:9]: Yeah that should be the case, shall I remove it? It was there before I edited the code

**Comment by akbarkz** [comment]
@immortalcodes The alignment of this section on small screen doesn't look right

**Comment by lyubomir-popov** [comment]
pls remove padding on the list items:

image

- hrs should have is-muted unless highlgihted or atr the beginning of each section:

image

- pls use vanilla's new [image background utilities ](https://vanillaframework.io/docs/patterns/images) to add a grey background here:

image

- same here 

image

- pls remove the padding-top here:

image

- pls follow this design for trhe medium/small screen layouts of this:
image
image

**Comment by lyubomir-popov** [comment]
last thing - p-section--shallow on these please:
image

**Review by akbarkz — APPROVED** [review]

**Pull request merged**